pkg/types/baremetal: use camelCase for JSON keys

stbenjam opened the pull request:
The variables are APIVIP and IngressVIP, which should use apiVIP and
ingressVIP for their JSON keys, respectively.

markmc commented:
> pkg/types/baremetal: use apiVIP and ingressVIP as json keys
>
> The variables are APIVIP and IngressVIP, which should use apiVIP and
> ingressVIP for their JSON keys, respectively.

A good habit to get into is to explain the "why" in commit messages, not just the what

Why should we use camleCase? I guess because we follow Kubernetes naming conventions https://github.com/eBay/Kubernetes/blob/master/docs/devel/api-conventions.md#naming-conventions

> Go field names must be CamelCase. JSON field names must be camelCase. Other than capitalization of the initial letter, the two should almost always match. No underscores nor dashes in either.

markmc commented:
Take a look at this:

```
$> git grep -n '`json:"' pkg/ | grep -v terraform | grep -v 'json:"[a-z][a-zA-Z0-9]*[,"]' | grep -v tfvars
...
pkg/types/baremetal/metadata.go:5:	LibvirtURI string `json:"libvirt_uri"`
pkg/types/baremetal/metadata.go:6:	IronicURI  string `json:"ironic_uri"`
pkg/types/baremetal/platform.go:34:	LibvirtURI string `json:"libvirt_uri,omitempty"`
pkg/types/baremetal/platform.go:39:	IronicURI string `json:"ironic_uri,omitempty"`
pkg/types/baremetal/platform.go:43:	ExternalBridge string `json:"external_bridge,omitempty"`
pkg/types/baremetal/platform.go:47:	ProvisioningBridge string `json:"provisioning_bridge,omitempty"`
pkg/types/baremetal/platform.go:62:	APIVIP string `json:"api_vip"`
pkg/types/baremetal/platform.go:65:	IngressVIP string `json:"ingress_vip"`
...
```

We should fix all of those

stbenjam committed:
pkg/types/baremetal: use camelCase for JSON keys

The installer follows Kubernetes naming conventions. This changes
the baremetal platform and metadata to use camelCase instead of snake
case per the guidelines found at [1].

[1] https://github.com/eBay/Kubernetes/blob/master/docs/devel/api-conventions.md#naming-conventions

stbenjam commented:
@markmc Thanks, I've updated the PR. 

The only of these fields dev-scripts sets is apiVIP and according to the [comment here](https://github.com/openshift-metal3/kni-installer/pull/143#issuecomment-509796651) it may not even be needed as the default behavior looks up the value from DNS.  Let's see what CI says.

markmc commented:
Looks good to me, feel free to merge after CI passes

metal3ci commented:
Build SUCCESS, see build http://10.8.144.11:8080/job/dev-tools/869/
